File: frontend/src/types/AnswerTemplate.ts
// src/types/AnswerTemplate.ts

export interface TemplateOption {
  id: number;
  text: string;
  order: number;
}

export interface AnswerTemplate {
  id: number;
  name: string;
  optionCount: number;
  options: TemplateOption[];
}

export interface AnswerTemplateCreateDto {
  name: string;
  options: { text: string; order: number }[];
}

export interface AnswerTemplateUpdateDto extends AnswerTemplateCreateDto {
  id: number;
}

File: backend/services/SurveyApp.SurveyManagement/SurveyApp.SurveyManagement/Domain/Entities/AnswerTemplate.cs
using SurveyApp.Shared.Persistance.Entities;
using SurveyApp.SurveyManagement.Application.Dto_s;

namespace SurveyApp.SurveyManagement.Domain.Entities
{
    public class AnswerTemplate : IEntity
    {
        public int Id { get; set; }
        public string Name { get; set; }           
        public int OptionCount { get; set; }        
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        public ICollection<TemplateOption> Options { get; set; } = new List<TemplateOption>();
        public ICollection<Question> Questions { get; set; } = new List<Question>();
    }

}
